src/main/java/uk/gov/hmcts/reform/cpo/service/impl/CasePaymentOrdersServiceImpl.java

```diff
@@ -2,75 +2,67 @@
+    }
 
-        if (casePaymentOrderQueryFilter.isItAnEmptyCriteria()) {
-            return Page.empty();
-        }
-        validateCasePaymentOrderQueryFilter(casePaymentOrderQueryFilter);
 
-        final PageRequest pageRequest = getPageRequest(casePaymentOrderQueryFilter);
+    @Override
+    public Page<CasePaymentOrder> getCasePaymentOrders(final CasePaymentOrderQueryFilter casePaymentOrderQueryFilter) {
+
+        final Page<CasePaymentOrderEntity> casePaymentOrderEntities;
+        final PageRequest pageRequest = casePaymentOrderQueryFilter.getPageRequest();
         if (casePaymentOrderQueryFilter.isACasesIdQuery()) {
-            return casePaymentOrdersRepository.findByCaseIdIn(
-                casePaymentOrderQueryFilter.getListOfLongCasesIds(),
-                pageRequest
-            );
+            casePaymentOrderEntities = casePaymentOrdersRepository.findByCaseIdIn(
+                casePaymentOrderQueryFilter.getListOfLongCasesIds(), pageRequest);
         } else {
-            return casePaymentOrdersRepository.findByIdIn(
+            casePaymentOrderEntities = casePaymentOrdersRepository.findByIdIn(
                 casePaymentOrderQueryFilter.getListUUID(),
                 pageRequest
             );
         }
+        return getPageOfCasePaymentOrder(casePaymentOrderEntities, pageRequest);
     }
 
-    @Transactional
-    @Override
-    public CasePaymentOrder updateCasePaymentOrder(UpdateCasePaymentOrderRequest request) {
-        throw new UnsupportedOperationException("Implement me: see CPO-6");
-    }
+    private Page<CasePaymentOrder> getPageOfCasePaymentOrder(Page<CasePaymentOrderEntity> casePaymentOrderEntities,
+                                                             PageRequest pageRequest) {
 
-    private PageRequest getPageRequest(CasePaymentOrderQueryFilter casePaymentOrderQueryFilter) {
-        final List<Sort.Order> orders = new ArrayList<>();
-        orders.add(new Sort.Order(Sort.Direction.ASC, CasePaymentOrderQueryFilter.CASES_ID));
-        orders.add(new Sort.Order(Sort.Direction.ASC, CasePaymentOrderQueryFilter.ORDER_REFERENCE));
-        return PageRequest.of(
-            casePaymentOrderQueryFilter.getPageNumber(),
-            casePaymentOrderQueryFilter.getPageSize(),
-            Sort.by(orders)
-        );
-    }
+        final List<CasePaymentOrder> casePaymentOrders =
+            casePaymentOrderMapper.map(casePaymentOrderEntities.getContent());
 
-    private void validateCasePaymentOrderQueryFilter(final CasePaymentOrderQueryFilter casePaymentOrderQueryFilter) {
-        if (casePaymentOrderQueryFilter.isAnIdsAndCasesIdQuery()) {
-            throw new CasePaymentOrdersQueryException(
-                "case-payment-orders cannot filter case payments orders by both id and cases-id.");
-        }
+        return new PageImpl<>(casePaymentOrders, pageRequest,
+                              casePaymentOrderEntities.getTotalElements()
+        );
```

Comment:
this can all be simplified by using the `map` method on the `Page` class:
"Returns a new Page with the content of the current one mapped by the given Function."

Reply:
TODO